community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundQueryContext.scala

```diff
@@ -642,25 +656,36 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
       case e: NotFoundException => throw new EntityNotFoundException(s"Relationship with id $id", e)
     }
 
-    override def getByIdIfExists(id: Long): Option[RelationshipValue] = try {
-      var relationship: RelationshipProxy = null
-      transactionalContext.statement.readOperations().relationshipVisit(id, new RelationshipVisitor[Exception] {
-        override def visit(relationshipId: Long, typeId: Int, startNodeId: Long, endNodeId: Long): Unit = {
-          relationship = entityAccessor.newRelationshipProxy(relationshipId, startNodeId, typeId, endNodeId)
-        }
-      })
-      Option(fromRelationshipProxy(relationship))
-    } catch {
-      case _: exceptions.EntityNotFoundException => None
-    }
+    override def getByIdIfExists(id: Long): Option[RelationshipValue] =
+      if (reads().relationshipExists(id))
+        Some(fromRelationshipProxy(entityAccessor.newRelationshipProxy(id)))
```

Comment:
Here I preferred the previous impl which cached the start/end/type of the relationship. That stuff is going to be in L1 cache anyway (given current store layout) so we might as well return it.